common/lib/xmodule/xmodule/partitions/partitions.py

```diff
@@ -290,8 +290,9 @@ def get_partition_from_id(partitions, user_partition_id):
     Returns:
         A UserPartition, or None if not found.
     """
+    if not partitions:
```

Comment:
So just to clarify here, this check is necessary because `partitions` could be passed in as a `None` rather than `[]`? Because if the `None` case does not exist, these two lines can be dropped. Feels odd to me because we didn't have this check before and I don't think we have seen errors because of it.

Reply:
Yes that's correct. There were no errors before because the test cases were always being skipped.

Reply:
Ah okay, I understand. My bad for missing this during your original round of comments. It was not clicking in my head. Please let me know when to re-review after tests and quality fixes are in place.